packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1360,3 +1360,39 @@ class KeepAlive extends ParentDataWidget<KeepAliveParentDataMixin> {
     properties.add(DiagnosticsProperty<bool>('keepAlive', keepAlive));
   }
 }
+
+/// A sliver that constrains the cross axis extent of its sliver child.
+///
+/// The [SliverConstrainedCrossAxis] takes an [extent] parameter and uses it as
+/// the cross axis extent of the [SliverConstraints] passed to the sliver child.
+/// The widget ensures that the [extent] is a value between 0 and the parent's
+/// cross axis extent.
+///
+/// This is useful when you want to apply a custom cross-axis extent constraint
```

Comment:
This is probably a good spot to mention that slivers typically consume the full cross axis extent.

Reply:
Can you add some sample code for using this sliver? This is an example from SliverFillRemaining: https://github.com/flutter/flutter/blob/master/examples/api/lib/widgets/sliver_fill/sliver_fill_remaining.0.dart

Reply:
Hey I just added the sample code, but was wondering if there is an easy way to test it, and what I should do with it.

Reply:
The sample should be embedded here like this:

https://github.com/flutter/flutter/blob/4c3d7333b174f4f061f4d871c9e97b8950efcd74/packages/flutter/lib/src/widgets/sliver_fill.dart#L205

And in order to test it, there is a test directory in examples/api/test
Each sample has it's own test file with the same naming convention.

Here is an example:

https://github.com/flutter/flutter/blob/master/examples/api/test/widgets/sliver/sliver_opacity.1_test.dart

Reply:
Done!